innersource-license.md

```diff
@@ -0,0 +1,83 @@
+## Title
+
+InnerSource License
+
+## Patlet
+
+Two companies that belong to the same group want to share software source code with each other but they are concerned about the implications in terms of legal liabilities or cross-company accounting. An **InnerSource License** provides a reusable legal framework for the inter-company agreement within the group, which opens up new collaboration options.
+
+## Problem
+
+When two companies within a group want to share code with each other, they need an agreement about the terms and often a legal contract. Creating such agreements on a per project basis takes effort and creates a barrier for sharing. i.e. a team/company might decide not to share their source code with another company in the group because it seems complicated.
+
+Barriers for sharing can lead to silos and duplication of effort in rebuilding similar solutions in multiple parts of the group.
+
+At the time of sharing the source code, it can not be reliably predicted what the value of sharing will be. Hence if the activity of sharing has any significant cost, the companies are unlikely to do it, as they are concerned about the return on investment.
+
+## Context
+
+- A large group with many companies (subsidiaries) that want to share code. (when the group gets larger, the value of this pattern increases)
+- These companies are legally independent units.
+- Multiple of these companies/subsidiaries are developing software, and are using services of the other companies. They have a motivation to contribute to each other’s source code.
+- Certain size of developers (crowdsourcing and ecosystem effects of InnerSource)
```

Comment:
```suggestion
```

Doesn't that apply to InnerSource in general? Is it a contextual factor only important for this pattern?

Reply:
(Mental note to ourselves: Maybe we need to define a general context (some kind of an `AbstractSuperContext`) for all patterns)

Reply:
I agree that there are likely some general points that always apply when an InnerSource approach is considered by a company.

For the time being I would like to keep this in here though, as we don't have such a general **InnerSource Context** (`AbstractSuperContext`) description yet. My assumption is that the readers need to be able to "read the pattern on it's own". 

But I can see that if we ever want to turn all proven patterns into a book, there would be an intro section that describes the **InnerSource Context**.

Do you want to create an issue to describe that **InnerSource Context** (`AbstractSuperContext`)? @gruetter could be interested in that. At the summit the topic of "how large does a company have to be for InnerSource to make sense" came up in a breakout that we were in.

@maxcapraro please close this conversation if you are fine with leaving this section as is.

Reply:
I think the specific condition for an Inner Source License is that the organization has a complexity so that there are legal contracts necessary within the organization. That probably corresponds with size and numbers of developers in most cases.

Reply:
>  [...] AbstractSuperContext [...]

Agreed, @spier! That's a topic for elsewhere :)

> I think the specific condition for an Inner Source License is that the organization has a complexity so that there are legal contracts necessary within the organization. That probably corresponds with size and numbers of developers in most cases.

Cool. It doesn't seem to be merely the size then. A possible fix could mabye be something like that? :)

```suggestion
- A sufficient complexity of the organization and its organizational structure
```